### ISSUE TITLE
Re-instantiating Storage Class is SLOW

### ISSUE THREAD

**Issue by jamesmfriedman** [jazzband/sorl-thumbnail#301]
the Storage class located in default.py is causing a lot of slow down. It reinitializes the storage class for every thumbnail that is requested. The problem with that is if you're using something like s3boto, it re-establishes a connection to s3 everytime.

**Comment by pior** [comment]
With Boto3 this issue is a real blocker because Boto3 loads some very large JSON files when creating a session object. On Heroku it means 100~150ms per thumbnail, only to generate the secure url (no network calls)

```shell
2017-03-25T03:19:40.384783+00:00 app[web.1]: 2017-03-24 23:19:40,384 DEBUG sorl.thumbnail.base:get_thumbnail:68 Getting thumbnail for file [sponsors/YulDev2_1.png] at [200x100]
2017-03-25T03:19:40.391243+00:00 app[web.1]: 2017-03-24 23:19:40,391 DEBUG botocore.loaders:load_file:174 Loading JSON file: /app/.heroku/python/lib/python2.7/site-packages/boto3/data/s3/2006-03-01/resources-1.json
2017-03-25T03:19:40.400819+00:00 app[web.1]: 2017-03-24 23:19:40,400 DEBUG botocore.loaders:load_file:174 Loading JSON file: /app/.heroku/python/lib/python2.7/site-packages/botocore/data/endpoints.json
2017-03-25T03:19:40.424714+00:00 app[web.1]: 2017-03-24 23:19:40,424 DEBUG botocore.loaders:load_file:174 Loading JSON file: /app/.heroku/python/lib/python2.7/site-packages/botocore/data/s3/2006-03-01/service-2.json
2017-03-25T03:19:40.535902+00:00 app[web.1]: 2017-03-24 23:19:40,535 DEBUG botocore.loaders:load_file:174 Loading JSON file: /app/.heroku/python/lib/python2.7/site-packages/botocore/data/_retry.json
...
```

**Comment by claudep** [comment]
Should be fixed by #719
